src/scalarstats.jl

```diff
@@ -37,6 +37,26 @@ function harmmean(a::RealArray)
     return n / s
 end
 
+# Generalized mean
+"""
+    genmean(a, p)
+Return the generalized/power mean with exponent p of a real-valued array.
+"""
+function genmean(a::RealArray, p::Real)
+  if p<0
+    p = convert(AbstractFloat, p)# Negative int p causes domain error.
+  end
+  if p==0.0
+    return geomean(a)
+  end
+  s = 0.0
+  n = length(a)
+  for i = 1:n
```

Comment:
Most likely not. That shouldn't make a difference in practice. Keep it that way if you prefer consistency.